subsys/llext/llext_link.c

```diff
@@ -159,18 +208,23 @@ static void llext_link_plt(struct llext_loader *ldr, struct llext *ext,
 
 		switch (stb) {
 		case STB_GLOBAL:
+			/* First try the global symbol table */
 			link_addr = llext_find_sym(NULL,
 				SYM_NAME_OR_SLID(name, sym_tbl.st_value));
 
 			if (!link_addr) {
+				/* Next try internal tables */
 				link_addr = llext_find_sym(&ext->sym_tab, name);
 			}
 
 			if (!link_addr) {
+				/* Finally try any loaded tables */
 				struct llext_sym_ext se = {.sym = name};
 				ret = llext_iterate(llext_get_symbol, &se);
```

Comment:
I see why we need ext added to the llext_sym_ext now, but this still seems like the wrong place for the added member.

Reply:
+1

Would be OK to me if names were different (e.g., `struct llext_sym_ext` -> `struct global_sym_search_workbuf` and `llext_get_symbol` -> `search_symbol_by_name`); but really, I'd rather see this wrapped in a `int llext_search_sym(const void *name, struct llext **exporting_module)` function (which could even be made public? although IDK if it would be very useful...)

Reply:
> I see why we need ext added to the llext_sym_ext now, but this still seems like the wrong place for the added member.

@teburd sorry, don't quite understand. Why is this a wrong place? This structure is only local for this sole iteration run. It's introduced in this very PR a few commits earlier. Why cannot I add members to it?

Reply:
> +1
> 
> Would be OK to me if names were different (e.g., `struct llext_sym_ext` -> `struct global_sym_search_workbuf` and `llext_get_symbol` -> `search_symbol_by_name`); but really, I'd rather see this wrapped in a `int llext_search_sym(const void *name, struct llext **exporting_module)` function (which could even be made public? although IDK if it would be very useful...)

@mathieuchopstm I could make it a function, sure, probably `static` for now, but don't think that structure deserves any "global" designation in its name - it's perfectly local and confined to this single use only

Reply:
The "global" in my suggestion was meant to say "global search (for symbols)", as in "search for symbols everywhere". A more appropriate suggestion name suggestion would have been `llext_sym_search_param`.

IMO, the `llext_` prefix in iterate callback is confusing (sounds like an API you could call); and the structure name is cryptic which makes it confusing as to what its role is.

I think `int llext_search_sym()` could be useful in some situations (as `llext_iterate()` currently is :smile:), but I'm completely fine with just renaming.